src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.SymbolToProjectId.cs

```diff
@@ -78,7 +78,7 @@ internal partial class SolutionState
                 if (result != null)
                     return result;
 
-                symbol = symbol.ContainingSymbol;
+                symbol = symbol.ContainingSymbol!;
```

Comment:
Consider extracting the loop variable to an `ISymbol? current` local. The current pattern feels unsafe.